catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -69,45 +75,59 @@ public CatalogBackupPlugin() {
      * Backs up created metacards to the file system backup.
      *
      * @param input the {@link CreateResponse} to process
-     * @return
+     * @return {@link CreateResponse}
      * @throws PluginExecutionException
      */
     @Override
     public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+
         if (enableBackupPlugin && Requests.isLocal(input.getRequest())) {
-            LOGGER.debug("Performing backup of metacards in CreateResponse.");
+            LOGGER.trace("Performing backup of metacards in CreateResponse.");
 
             if (rootBackupDir == null) {
                 throw new PluginExecutionException("No root backup directory configured.");
             }
 
-            List<String> errors = new ArrayList<String>();
+            EXECUTOR_SERVICE.submit(() -> processBatch(input.getCreatedMetacards()));
```

Comment:
So what I was expecting here is to see these created metacards getting dumped into a queue. Then you'd have another thread waiting until that queue hit a particular size or some period of time passed since the last dump of the queue. Then you'd pull everything out of the queue and process that batch. If you do it like this, you could end up still working on one metacard at a time which would exacerbate poor performance on someone's already terrible ingest.